---
title: "Lab 8 Homework"
author: "Dawson Diaz"
date: "`r Sys.Date()`"
output:
  html_document:
    theme: spacelab
    toc: no
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


### Libraries


```{r}
#install.packages("ape")
#install.packages("seqinr")
#install.packages("rentrez")
```



```{r message=FALSE, warning=FALSE}
library(ape)
library(seqinr)
library(rentrez)
```
<br>

**1. All the databases associated with NCBI are updating and sharing their data daily. With the new strain of the coronavirus, SARS-CoV-2 at the forefront of a lot of medical research, genomes have already been uploaded to GenBank. The accession number for a complete genome of an isolate from China is "MT093631.1". Pull this sequence from GenBank into R, then write it as a .fasta file and read the .fasta file into R using the code we learned in class.**

Pull from GenBank
```{r}
CoV19 <- read.GenBank("MT093631.1", as.character = TRUE)
```
<br>
Write `.fasta`
```{r}
write.fasta(names="CoV19", sequences=CoV19, file.out="data/CoV-19.fasta")
```

<br>
Read `.fasta`
```{r}
CoV19 <- read.fasta(file = "data/CoV-19.fasta")
```

<br>
Extract Sequence
```{r}
CoV19 <-CoV19[[1]]
```


<br>
**2. Explore this sequence using functions we learned in class. What is the length of the genome? Create a table showing the nucleotide composition of the sequence, and calculate the GC content.**

```{r}
length(CoV19)
```
<br>
```{r}
table(CoV19)
```

<br>
```{r}
GC(CoV19)
```



<br>
**3. Several genomes for the new strain of coronavirus have already been uploaded to GenBank. Using search terms, look for uploaded genomes SARS-CoV-2. (Hint: You're looking for things listed as genomes, which you won't find looking for a [GENE] keyword 'entrez_db_searchable("nuccore")' will provide you with a list of search term options.) Limit your returned hits to 10. Download your ten genomes and write them out as a .fasta file.**

Search for sequences
```{r}
CoV19_search <- entrez_search(db="nuccore", term= "SARS-CoV-2[ORGN]", retmax=10)
```

<br>
Fetch those sequences
```{r}
CoV19_seqs <- entrez_fetch(db="nuccore", id=CoV19_search$ids, rettype="fasta")
```
<br>

Write to `.fasta`
```{r}
write.fasta(names="CoV19", sequences=CoV19_seqs, file.out="data/CoV19_seqs.fasta")
```




<br><br><br>